src/stat.go

```diff
@@ -125,27 +150,79 @@ func (g *Commands) stat(relToRootPath string, file *File, depth int) error {
 	}
 
 	// Arbitrary value for throttle pause duration
+	// TODO Is this really needed now that everything is serialized?
+
 	throttle := time.Tick(1e9 / 5)
 
-	prettyFileStat(g.log.Logf, relToRootPath, file)
-	perms, permErr := g.rem.listPermissions(file.Id)
-	if permErr != nil {
-		return permErr
-	}
+	if !g.opts.Md5sum {
+		prettyFileStat(g.log.Logf, relToRootPath, file)
+		perms, permErr := g.rem.listPermissions(file.Id)
+		if permErr != nil {
+			return permErr
+		}
 
-	for _, perm := range perms {
-		prettyPermission(g.log.Logf, perm)
+		for _, perm := range perms {
+			prettyPermission(g.log.Logf, perm)
+		}
+	} else if file.Md5Checksum != "" {
+		g.log.Logf("%32s  %s\n", file.Md5Checksum, strings.TrimPrefix(relToRootPath, "/"))
 	}
 
-	if !file.IsDir {
-		return nil
-	}
+	if file.IsDir {
+		//remoteChildren := FileArray{}
+		var remoteChildren FileArray
+
+		for child := range g.rem.FindByParentId(file.Id, g.opts.Hidden) {
+			remoteChildren = append(remoteChildren, child)
+			<-throttle
+		}
 
-	remoteChildren := g.rem.FindByParentId(file.Id, g.opts.Hidden)
-	for child := range remoteChildren {
-		g.stat(filepath.Clean(relToRootPath+"/"+child.Name), child, depth)
-		<-throttle
+		if g.opts.Md5sum {
+			// TODO use g.sort instead of sort.stable
+			// i.e g.sort(remoteChildren,"name")
+			// The reason this is not done here is because g.sort does not sort in natural order
+			
+			sort.Stable(remoteChildren)
+		}
+
+		for _, child := range remoteChildren {
+			g.stat(filepath.Clean(relToRootPath+"/"+child.Name), child, depth)
+		}
 	}
 
 	return nil
 }
+
```

Comment:
Once you've created the specialized function and moved this code to src/sort.go then you'll be able to just say 
```go
g.sort(remoteChildren, NameKey)
```